addons/mail/static/src/models/chat_window_manager/chat_window_manager.js

```diff
@@ -137,7 +137,9 @@ function factory(dependencies) {
             _newOrdered[index] = otherChatWindow.localId;
             _newOrdered[index + 1] = chatWindow.localId;
             this.update({ _ordered: _newOrdered });
-            chatWindow.focus();
+            if (!chatWindow.isFolded) {
+                chatWindow.focus();
+            }
```

Comment:
Looks good, though from my testing the traceback happens due to bad code in components of managing `.focus()` (something to do with attempting to focus the thread viewer component, which doesn't exist).

This code is good in models, but I would also fix the bad code in components.

Reply:
You can see the problem in two different ways : 
- it's wrong to try to focus a folded chat window
- when chat window is folded and we try to focus it, then it should not try to focus on unexisting elements

In my way of thinking the first one is the real problem that should be avoided and so fixing the second would only lead to situation where we want to badly focus something that should not be. So I didn't change anything in chat window component. Do you think we still have to ?

Reply:
focus on a folded chat window makes sense, like imagine we interact with chat windows solely with keyboard navigation (partial support ATM). So good management of focus of folded chat window by the component itself seems appropriate, even though that's no big deal now.

Reply:
Considering a folded chat window can be focused, then my fix is wrong and instead I should handle the focus in chat window component when it's folded

Reply:
Note that if folded chat windows can be focused now, we should probably change tab cycling as well, because it currently goes to the "next visible unfolded", instead of just "next".

Reply:
> Note that if folded chat windows can be focused now, we should probably change tab cycling as well, because it currently goes to the "next visible unfolded", instead of just "next".

Would say out of scope of PR, and the current UI feedback on active chat window is not good enough to consider folded chat windows in TAB.